src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,153 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<E> {
+        private E item;
+        private Node<E> next;
+        private Node<E> prev;
+
+        Node(Node<E> prev, E element, Node<E> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            addFirst(value);
+        } else {
+            addLast(value);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index out of bound: " + index);
+        }
+        if (isEmpty()) {
+            addFirst(value);
+        } else if (index == size) {
+            addLast(value);
+        } else if (index == 0) {
+            head.prev = new Node<>(null, value, head);;
+            head = head.prev;
+        } else {
+            Node<T> current = getNodeByIndex(index);
+            Node<T> newNode = new Node<>(current.prev, value, current);
+            current.prev.next = current.prev = newNode;
+        }
+        size++;
```

Comment:
Here we may more actively reuse existing method to simplify code
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        checkIndex(index);
        if (index == 0) {
            head.prev = new Node<>(null, value, head);;
            head = head.prev;
        } else {
            Node<T> current = getNodeByIndex(index);
            Node<T> newNode = new Node<>(current.prev, value, current);
            current.prev.next = current.prev = newNode;
        }
        size++;
```